src/rgw/rgw_rest_sts.cc

```diff
@@ -148,59 +148,52 @@ WebTokenEngine::is_cert_valid(const vector<string>& thumbprints, const string& c
 
 template <typename T>
 void
-WebTokenEngine::recurse_and_insert(const string& key, const jwt::claim& c, T& t) const
+WebTokenEngine::recurse_and_insert(const string& key, const jwt::basic_claim<traits>& c, T& t) const
 {
   string s_val;
-  jwt::claim::type c_type = c.get_type();
+  auto c_type = c.get_type();
   switch(c_type) {
-    case jwt::claim::type::null:
-      break;
-    case jwt::claim::type::boolean:
-    case jwt::claim::type::number:
-    case jwt::claim::type::int64:
+    case jwt::json::type::boolean:
+    case jwt::json::type::number:
+    case jwt::json::type::integer:
     {
       s_val = c.to_json().serialize();
       t.emplace(std::make_pair(key, s_val));
       break;
     }
-    case jwt::claim::type::string:
+    case jwt::json::type::string:
     {
       s_val = c.to_json().to_str();
       t.emplace(std::make_pair(key, s_val));
       break;
     }
-    case jwt::claim::type::array:
+    case jwt::json::type::array:
     {
-      const picojson::array& arr = c.as_array();
+      const auto& arr = c.as_array();
       for (auto& a : arr) {
-        recurse_and_insert(key, jwt::claim(a), t);
+        recurse_and_insert(key, jwt::basic_claim<traits>(a), t);
       }
       break;
     }
-    case jwt::claim::type::object:
-    {
-      const picojson::object& obj = c.as_object();
-      for (auto& m : obj) {
```

Comment:
@awojno-bloomberg : why was this case removed?

Reply:
looking through their github, i found that their [FAQ](https://thalhammer.github.io/jwt-cpp/md_docs_2faqs.html) has a section:
> Why does [jwt::basic_claim](https://thalhammer.github.io/jwt-cpp/classjwt_1_1basic__claim.html) have no `as_object()` method?

that suggests using `to_json()` instead

Reply:
@awojno-bloomberg we weren't able to take your commit because we rely on this support for nested claims (see https://github.com/ceph/ceph/pull/61796#issuecomment-2812130912). can you please revisit this part in https://github.com/ceph/ceph/pull/60985?